[Build fails] multiple definition of `renderer_sync'

## Description

if you build pcsx_rearmed for ARM devices with `BUILTIN_GPU=neon` enabled `THREAD_RENDERING=1` you end up with `multiple definition of `renderer_sync'` and else see log:

https://pastebin.com/feVUaeLW

## Steps to reproduce

package:
https://github.com/SupervisedThinking/LibreELEC-RR/blob/master-rr/packages/supervisedthinking/emulation/libretro/pcsx_rearmed/package.mk

## When did the behavior start?

I'm pretty sure it's related to https://github.com/libretro/pcsx_rearmed/pull/462 which was only tested for GPU=unai
## Your device/OS/platform/architecture

LibreELEC / ARM neon builds

@justinweiss this patch compiles fine, haven't runtested yet:
```
--- a/Makefile
+++ b/Makefile
@@ -194,9 +194,11 @@ plugins/gpu_unai/gpulib_if.o: CFLAGS += -DREARMED -O3
 CC_LINK = $(CXX)
 endif
 ifeq "$(THREAD_RENDERING)" "1"
+ifneq "$(BUILTIN_GPU)" "neon"
 CFLAGS += -DTHREAD_RENDERING
 OBJS += plugins/gpulib/gpulib_thread_if.o
 endif
+endif
```

The other one fails:
```
/home/supervisedthinking/development/LibreELEC-RR/build.LibreELEC-RPi4.arm-9.80-devel/toolchain/bin/armv8a-libreelec-linux-gnueabihf-gcc -march=armv8-a+crc -mtune=cortex-a53 -mabi=aapcs-linux -Wno-psabi -Wa,-mno-warn-deprecated -mfloat-abi=hard -mfpu=neon-fp-armv8 -Wall -pipe  -O2 -fomit-frame-pointer -DGIT_VERSION=\"976907a\" -marm -fPIC -D_FILE_OFFSET_BITS=64 -Wall -Iinclude -ffast-math -O2 -DNDEBUG -Ideps/zlib -DNEW_DYNAREC -DGPU_NEON -DTHREAD_RENDERING -Ideps/libchdr -DHAVE_SYS_PARAM_H -Ideps/crypto -Ideps/flac-1.3.2/include -Ideps/flac-1.3.2/src/libFLAC/include -Ideps/flac-1.3.2/src/libFLAC/include -Ideps/lzma-16.04/C -DHAVE_CHD -D'PACKAGE_VERSION="1.3.2"' -DFLAC__HAS_OGG=0 -DFLAC__NO_DLL -DHAVE_LROUND -DHAVE_STDINT_H -DHAVE_STDLIB_H -DFLAC__NO_DLL -D_7ZIP_ST -Ilibretro-common/include -DFRONTEND_SUPPORTS_RGB565 -DHAVE_LIBRETRO -DNO_FRONTEND -DNEON_BUILD -DTEXTURE_CACHE_4BPP -DTEXTURE_CACHE_8BPP -c -o plugins/gpu_neon/psx_gpu_if.o plugins/gpu_neon/psx_gpu_if.c
plugins/gpu_neon/psx_gpu_if.c: In function 'real_do_cmd_list':
plugins/gpu_neon/psx_gpu_if.c:46:17: error: 'scratch_ex_regs' undeclared (first use in this function)
   46 | #define ex_regs scratch_ex_regs
      |                 ^~~~~~~~~~~~~~~
plugins/gpu_neon/psx_gpu_if.c:60:3: note: in expansion of macro 'ex_regs'
   60 |   ex_regs[1] &= ~0x1ff;
      |   ^~~~~~~
plugins/gpu_neon/psx_gpu_if.c:46:17: note: each undeclared identifier is reported only once for each function it appears in
   46 | #define ex_regs scratch_ex_regs
      |                 ^~~~~~~~~~~~~~~
plugins/gpu_neon/psx_gpu_if.c:60:3: note: in expansion of macro 'ex_regs'
   60 |   ex_regs[1] &= ~0x1ff;
      |   ^~~~~~~
plugins/gpu_neon/psx_gpu_if.c: In function 'real_renderer_init':
plugins/gpu_neon/psx_gpu_if.c:46:17: error: 'scratch_ex_regs' undeclared (first use in this function)
   46 | #define ex_regs scratch_ex_regs
      |                 ^~~~~~~~~~~~~~~
plugins/gpu_neon/psx_gpu_if.c:106:3: note: in expansion of macro 'ex_regs'
  106 |   ex_regs = gpu.ex_regs;
      |   ^~~~~~~
make[1]: *** [Makefile:383: plugins/gpu_neon/psx_gpu_if.o] Error 1
```

Anyway the first patch "fakes threaded" rendering but opt it out in the `Makefile` for `GPU=neon` so people will assume it's enabled if they have a look in the `makefile.libretro` then? I'm not sure if this is a proper solution 🤔

Latest patch:

```
/home/supervisedthinking/development/LibreELEC-RR/build.LibreELEC-RK3399.arm-9.80-devel/toolchain/bin/armv8a-libreelec-linux-gnueabihf-gcc -march=armv8-a+crc -mtune=cortex-a72.cortex-a53 -mabi=aapcs-linux -Wno-psabi -Wa,-mno-warn-deprecated -mfloat-abi=hard -mfpu=crypto-neon-fp-armv8 -Wall -pipe  -O2 -fomit-frame-pointer -DGIT_VERSION=\"976907a\" -marm -fPIC -D_FILE_OFFSET_BITS=64 -Wall -Iinclude -ffast-math -O2 -DNDEBUG -Ideps/zlib -DNEW_DYNAREC -DGPU_NEON -DTHREAD_RENDERING -Ideps/libchdr -DHAVE_SYS_PARAM_H -Ideps/crypto -Ideps/flac-1.3.2/include -Ideps/flac-1.3.2/src/libFLAC/include -Ideps/flac-1.3.2/src/libFLAC/include -Ideps/lzma-16.04/C -DHAVE_CHD -D'PACKAGE_VERSION="1.3.2"' -DFLAC__HAS_OGG=0 -DFLAC__NO_DLL -DHAVE_LROUND -DHAVE_STDINT_H -DHAVE_STDLIB_H -DFLAC__NO_DLL -D_7ZIP_ST -Ilibretro-common/include -DFRONTEND_SUPPORTS_RGB565 -DHAVE_LIBRETRO -DNO_FRONTEND -c -o frontend/plugin.o frontend/plugin.c
In file included from plugins/gpu_neon/psx_gpu_if.c:20:
plugins/gpu_neon/../gpulib/gpulib_thread_if.h:27:23: error: unknown type name 'uint32_t'
   27 | int  real_do_cmd_list(uint32_t *list, int count, int *last_cmd);
      |                       ^~~~~~~~
plugins/gpu_neon/../gpulib/gpulib_thread_if.h:30:31: error: unknown type name 'uint32_t'
   30 | void real_renderer_sync_ecmds(uint32_t * ecmds);
      |                               ^~~~~~~~
plugins/gpu_neon/../gpulib/gpulib_thread_if.h:34:44: warning: 'struct rearmed_cbs' declared inside parameter list will not be visible outside of this definition or declaration
   34 | void real_renderer_set_config(const struct rearmed_cbs *config);
      |                                            ^~~~~~~~~~~
In file included from plugins/gpu_neon/psx_gpu_if.c:47:
plugins/gpu_neon/../gpulib/gpu.h:96:12: error: duplicate member 'scratch_ex_regs'
   96 |   uint32_t scratch_ex_regs[8]; // for threaded rendering
      |            ^~~~~~~~~~~~~~~
plugins/gpu_neon/psx_gpu_if.c:28:29: error: conflicting types for 'real_renderer_set_config'
   28 | #define renderer_set_config real_renderer_set_config
      |                             ^~~~~~~~~~~~~~~~~~~~~~~~
plugins/gpu_neon/../gpulib/gpu.h:120:6: note: in expansion of macro 'renderer_set_config'
  120 | void renderer_set_config(const struct rearmed_cbs *config);
      |      ^~~~~~~~~~~~~~~~~~~
In file included from plugins/gpu_neon/psx_gpu_if.c:20:
plugins/gpu_neon/../gpulib/gpulib_thread_if.h:34:6: note: previous declaration of 'real_renderer_set_config' was here
   34 | void real_renderer_set_config(const struct rearmed_cbs *config);
      |      ^~~~~~~~~~~~~~~~~~~~~~~~
plugins/gpu_neon/psx_gpu_if.c:28:29: error: conflicting types for 'real_renderer_set_config'
   28 | #define renderer_set_config real_renderer_set_config
      |                             ^~~~~~~~~~~~~~~~~~~~~~~~
plugins/gpu_neon/psx_gpu_if.c:187:6: note: in expansion of macro 'renderer_set_config'
  187 | void renderer_set_config(const struct rearmed_cbs *cbs)
      |      ^~~~~~~~~~~~~~~~~~~
In file included from plugins/gpu_neon/psx_gpu_if.c:20:
plugins/gpu_neon/../gpulib/gpulib_thread_if.h:34:6: note: previous declaration of 'real_renderer_set_config' was here
   34 | void real_renderer_set_config(const struct rearmed_cbs *config);
      |      ^~~~~~~~~~~~~~~~~~~~~~~~
make: *** [Makefile:383: plugins/gpu_neon/psx_gpu_if.o] Error 1
```

Technically you could clone my repo: https://github.com/SupervisedThinking/LibreELEC-RR and run:

`PROJECT=Rockchip DEVICE=RK3399 ARCH=arm scripts/build pcsx_rearmed`

--- a/Makefile
+++ b/Makefile
@@ -180,6 +180,10 @@ CFLAGS += -DGPU_PEOPS
 plugins/dfxvideo/gpulib_if.o: CFLAGS += -fno-strict-aliasing
 plugins/dfxvideo/gpulib_if.o: plugins/dfxvideo/prim.c plugins/dfxvideo/soft.c
 OBJS += plugins/dfxvideo/gpulib_if.o
+ifeq "$(THREAD_RENDERING)" "1"
+CFLAGS += -DTHREAD_RENDERING
+OBJS += plugins/gpulib/gpulib_thread_if.o
+endif
 endif
 ifeq "$(BUILTIN_GPU)" "unai"
 CFLAGS += -DGPU_UNAI
@@ -190,12 +194,12 @@ OBJS += plugins/gpu_unai/gpulib_if.o
 ifeq "$(ARCH)" "arm"
 OBJS += plugins/gpu_unai/gpu_arm.o
 endif
-plugins/gpu_unai/gpulib_if.o: CFLAGS += -DREARMED -O3 
-CC_LINK = $(CXX)
-endif
 ifeq "$(THREAD_RENDERING)" "1"
 CFLAGS += -DTHREAD_RENDERING
 OBJS += plugins/gpulib/gpulib_thread_if.o
+endif
+plugins/gpu_unai/gpulib_if.o: CFLAGS += -DREARMED -O3 
+CC_LINK = $(CXX)
 endif
 
 # cdrcimg

